applications/FluidDynamicsApplication/custom_utilities/fluid_test_utilities.h

```diff
@@ -0,0 +1,129 @@
+#include "containers/model.h"
+#include "includes/model_part.h"
+#include "processes/process.h"
+
+// Application includes
+
+namespace Kratos
+{
+///@name Classes
+///@{
+
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) FluidTestUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    using NodeType = ModelPart::NodeType;
+
+    using PropertiesType = ModelPart::PropertiesType;
+
+    using DofsVectorType =  std::vector<Dof<double>::Pointer>;
+
+    using EquationIdVectorType = std::vector<std::size_t>;
+
+    ///@}
+    ///@name Static operations
+    ///@{
+
+    template<class TDataType>
+    static void AssignRandomValues(
+        TDataType& rValue,
+        const std::string& rSeed,
+        const int DomainSize,
+        const double MinValue = 0.0,
+        const double MaxValue = 1.0);
+
+    static ModelPart& CreateTestModelPart(
+        Model& rModel,
+        const std::string& rModelPartName,
+        const std::string& rElementName,
+        const std::string& rConditionName,
+        const std::function<void(PropertiesType&)>& rSetElementProperties,
+        const std::function<void(PropertiesType&)>& rSetConditionProperties,
+        const std::function<void(ModelPart&)>& rAddNodalSolutionStepVariablesFuncion,
+        const std::function<void(NodeType&)>& rAddDofsFunction,
+        const int BufferSize = 2);
+
+    template<class TDataType>
+    static void RandomFillNodalHistoricalVariable(
+        ModelPart& rModelPart,
+        const Variable<TDataType>& rVariable,
+        const double MinValue = 0.0,
+        const double MaxValue = 1.0,
+        const int Step = 0);
+
+    template<class TContainerType, class TDataType>
+    static void RandomFillContainerNonHistoricalVariable(
+        TContainerType& rContainer,
+        const Variable<TDataType>& rVariable,
+        const IndexType DomainSize,
+        const double MinValue = 0.0,
+        const double MaxValue = 1.0);
+
+    ///@}
+    ///@name Classes
+    ///@{
+
+    template<class TContainerType>
+    class Testing
+    {
+    public:
+        ///@name Static Operations
+        ///@{
+
+        static void RunEntityGetDofListTest(
+            ModelPart& rModelPart,
+            const std::vector<const Variable<double>*>& rDofVariablesList);
+
+        static void RunEntityEquationIdVectorTest(
+            ModelPart& rModelPart,
+            const std::vector<const Variable<double>*>& rDofVariablesList);
+
+        static void RunEntityGetValuesVectorTest(
+            ModelPart& rModelPart,
+            const std::vector<const Variable<double>*>& rDofVariablesList);
+
+        static void RunEntityGetFirstDerivativesVectorTest(
+            ModelPart& rModelPart,
+            const std::function<Vector(const ModelPart::NodeType&)>& rValueRetrievalMethod);
+
+        static void RunEntityGetSecondDerivativesVectorTest(
+            ModelPart& rModelPart,
+            const std::function<Vector(const ModelPart::NodeType&)>& rValueRetrievalMethod);
```

Comment:
Instead of having the template for this internal `Testing` class, I'd overload all these methods for `Elements` and `Conditions` container. Then you can have an internal method templated to avoid reimplementing everything twice. Then the usage would be much easier.

Reply:
done 👍